library/Icinga/Chart/Primitive/Circle.php

```diff
@@ -61,8 +62,22 @@ public function toSvg(RenderContext $ctx)
         $circle->setAttribute('cx', Format::formatSVGNumber($coords[0]));
         $circle->setAttribute('cy', Format::formatSVGNumber($coords[1]));
         $circle->setAttribute('r', $this->radius);
-        $circle->setAttribute('style', $this->getStyle());
+
+        $this->id = $this->id ?? uniqid('circle-');
+        $circle->setAttribute('id', $this->id);
+
         $this->applyAttributes($circle);
+
+        $document = (new DOMDocument());
+        $document->loadHTML($this->getStyle());
+
+        $circle->appendChild(
+            $circle->ownerDocument->importNode(
+                $document->getElementsByTagName('style')->item(0),
+                true
+            )
+        );
```

Comment:
Please format your code like this in all the styleable classes you've changed.
```suggestion
        $style = (new DOMDocument())
            ->loadHTML($this->getStyle())
            ->getElementsByTagName('style')
            ->item(0);

        $circle->appendChild($circle->ownerDocument->importNode($style, true));
```

Reply:
As discussed `loadHTML()` returns `DOMDocument|bool` so method chaining might not be appropriate here.